123_state_tasks.R

```diff
@@ -17,19 +17,40 @@ do_state_tasks <- function(oldest_active_sites, ...) {
     }
   )
 
+  plot_step <- create_task_step(
+    step_name = 'plot',
+    target_name = function(task_name, ...) {
+      sprintf("3_visualize/out/timeseries_%s.png", task_name)
+    },
+    command = function(steps, ...) {
+      sprintf("plot_site_data(target_name, %s, parameter)",
+              steps[["download"]]$target_name)
+    }
+  )
+
+  tally_step <- create_task_step(
+    step_name = "tally",
+    target_name = function(task_name, ...) {
+      sprintf("%s_tally", task_name)
+    },
+    command = function(steps, ...) {
+      sprintf("tally_site_obs(%s)", steps[["download"]]$target_name)
```

Comment:
:+1:

Reply:
I'm still feeling mixed about whether it was worth implementing this `steps` option given that string manipulation really isn't that hard...but I'm glad you tried it.

Reply:
> given that string manipulation really isn't that hard

do you mean it would not be that difficult to just do `sprintf("tally_site_obs(%s_data)", task_name)` instead? I agree but I also kind of like this because then if I go change that target naming convention, I don't have to remember where I used it downstream, even if the actual change is trivial.

Reply:
yes, that's what i meant, and thanks for your input - that is why I did implement the `steps` option, so I'm glad you see value in it.